src/View.php

```diff
@@ -596,7 +599,9 @@ public function beginPage(): void
      */
     public function endPage(): void
     {
-        $this->eventDispatcher->dispatch(new PageEnd($this->getViewFile()));
+        if (is_string($viewFile = $this->getViewFile())) {
```

Comment:
How could it be non-string?

Reply:
> It could be false when rendering from View directly
> You can reproduce this behavior with the given method of the controller
> ` public function index(WebView $view, DataResponseFactoryInterface $responseFactory): ResponseInterface { return $responseFactory->createResponse($view->render('index')); }`

same, but probably only when called this method alone